### PR TITLE
fix: (2.35) Use encryption for SMS confidential parameters [DHIS2-10076]

### PR DESCRIPTION
**Pull request by zubaira**

### CODE REVIEW

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java [hunk 1]
```diff
@@ -107,4 +108,30 @@ public void setEncode( boolean encode )
     {
         this.encode = encode;
     }
+
+    @Override
+    public boolean equals( Object o )
```
Comment (gnespolino) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java:113]: what about using lombok's
```
@EqualsAndHashCode( of = {"key", "value", "header", "encode", "confidential"} )
```

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java [hunk 2]
```diff
@@ -50,6 +52,13 @@
 public class BulkSmsHttpGateway
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java:52]: Could remove constructor and annotate class with
```
@RequiredArgsConstructor
```

and 

```
@Qualifier( "tripleDesStringEncryptor" )
private final PBEStringEncryptor pbeStringEncryptor;
```

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/ClickatellHttpGateway.java [hunk 3]
```diff
@@ -50,6 +52,13 @@
 public class ClickatellHttpGateway
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/ClickatellHttpGateway.java:52]: same as BulkSmsHttpGateway

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java [hunk 4]
```diff
@@ -366,4 +385,18 @@ private void initializeSmsConfig()
 
         loadGatewayConfigurationMap( smsConfiguration );
     }
+
+
+    private boolean isPresent( List<GenericGatewayParameter> parameters, GenericGatewayParameter parameter )
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java:390]: might be easier to read using:
```java
        return parameters.stream()
                .anyMatch(parameter::equals);
```

**Review by gnespolino — APPROVED** [review]

**Pull request merged**